api/lib/source/polkadotV0-source.js

```diff
@@ -529,6 +534,166 @@ class polkadotAPI {
       }
     )
   }
+
+  async getAllProposals() {
+    const api = await this.getAPI()
+
+    const [
+      blockHeight,
+      totalIssuance,
+      democracyProposals,
+      democracyReferendums,
+      treasuryProposals,
+      councilProposals,
+      councilMembers
+    ] = await Promise.all([
+      this.getBlockHeight(),
+      api.query.balances.totalIssuance(),
+      api.derive.democracy.proposals(),
+      api.derive.democracy.referendums(),
+      api.derive.treasury.proposals(),
+      api.derive.council.proposals(),
+      api.query.council.members()
+    ])
+    const allProposals = democracyProposals
+      .map((proposal) => {
+        return this.reducers.democracyProposalReducer(
+          this.network,
+          proposal,
+          totalIssuance,
+          blockHeight
+        )
+      })
+      .concat(
+        democracyReferendums.map((proposal) => {
+          return this.reducers.democracyReferendumReducer(
+            this.network,
+            proposal,
+            totalIssuance,
+            blockHeight
+          )
+        })
+      )
+      .concat(
+        treasuryProposals.proposals.map((proposal) => {
+          return this.reducers.treasuryProposalReducer(
+            this.network,
+            proposal,
+            councilMembers
+          )
+        })
+      )
+      .concat(
+        councilProposals.map((proposal) => {
+          return this.reducers.councilProposalReducer(
+            this.network,
+            proposal,
+            councilMembers,
+            blockHeight
+          )
+        })
+      )
+
+    return orderBy(allProposals, 'id', 'desc')
```

Comment:
aren't the proposal IDs unique per type?

Reply:
yes, they are. What conflict do you see?

Reply:
if they are numbered by type than we can't order them by ID because it will not be chronological. 

```
4
4
4
3
3
2
2
2
1
1
1
1
```

Reply:
it will be chronological but with mixed types. So we should also order by type, not only id, good observation

Reply:
The id cant be the proposal Id but needs to be a combination of id and type as it would fuck with caching if multiple have the same id

Reply:
ouch true

Reply:
Done in #4507 and imported